### PR TITLE
Fix numpy1.19

### PR DESCRIPTION
**Pull request by ericpre**
Fix travis failure mentioned in #2430 due to the last version of numpy: https://github.com/numpy/numpy/releases/tag/v1.19.0

### Progress of the PR
- [x] fix axes_manager `next` (https://github.com/numpy/numpy/pull/15887),
- [x] fix `tostring` deprecation warning,
- [x] fix numpy ragged array deprecation warning,
- [x] fix complex change dtype deprecation warning,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/axes.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 class ndindex_nat(np.ndindex):
 
     def __next__(self):
-        return super(ndindex_nat, self).next()[::-1]
+        return super(ndindex_nat, self).__next__()[::-1]
```
Comment (francisco-dlp) [inline hyperspy/axes.py:52]: ``next(super(ndindex_nat, self))[::-1]``?

Comment (ericpre) [reply hyperspy/axes.py:52]: This is the first thing I have tried and it doesn't work...

Here is a minimal example to reproduce:
```python
import hyperspy.api as hs

s = hs.signals.Signal1D(np.arange(10*10*10).reshape((10, 10, 10)))
[coordinate for coordinate in s.axes_manager._am_indices_generator()]
```
```python
Traceback (most recent call last):

  File "<ipython-input-3-181577425fbe>", line 1, in <module>
    [coordinate for coordinate in s.axes_manager._am_indices_generator()]

  File "<ipython-input-3-181577425fbe>", line 1, in <listcomp>
    [coordinate for coordinate in s.axes_manager._am_indices_generator()]

  File "/home/eric/Dev/hyperspy/hyperspy/axes.py", line 52, in __next__
    return next(super(ndindex_nat, self))[::-1]

TypeError: 'super' object is not an iterator
```

Comment (francisco-dlp) [reply hyperspy/axes.py:52]: Interesting, it seems then that ``super`` is evaluated after ``next``.

**Comment by ericpre** [comment]
I made this PR to the wrong branch (minor instead of patch), but it was based on the patch branch, so I manually push these commit to the patch branch!

**Pull request merged**